packages/extension/src/ui/features/settings/DeveloperSettings.tsx

```diff
@@ -35,30 +37,54 @@ const DeveloperSettings: FC = () => {
       })
   }, [])
 
-  const onDiscoverAccounts = useCallback(async () => {
-    setLoading(true)
+  const discoveryLocalAccounts = useCallback(async () => {
     try {
       const allWalletAccounts = await getAccounts(true)
       const allWalletAccountsOnNetwork = accountsOnNetwork(allWalletAccounts, currentNetwork.id)
       const result = await discoverAccounts(currentNetwork.id)
       if (result === "error") {
-        console.log("Error discovering accounts")
-        setLoading(false)
+        console.log("Error discovering local accounts")
       } else {
         const discoveredAccounts = mapWalletAccountsToAccounts(result.accounts)
         let accountIndex = allWalletAccountsOnNetwork.length
         discoveredAccounts.forEach((account) => {
           setAccountName(account.networkId, account.address, getDefaultAccountNameByIndex(account, accountIndex))
           accountIndex++
         })
-        setLoading(false)
       }
     } catch (e) {
-      console.log("Error discovering accounts", e)
-      setLoading(false)
+      console.log("Error discovering local accounts", e)
     }
   }, [currentNetwork.id, setAccountName])
 
+  const discoveryLedgerAccounts = useCallback(async () => {
+    try {
+      const allWalletAccounts = await getAccounts(true)
```

Comment:
Q: Shall we check if ledger is connected?

Reply:
The ledger js sdk does not provide a method to check whether the device is connected. It will directly throw an exception if it is not connected, and we will ignore the error in such a case.

Reply:
I see, thanks!